src/backend/marsha/core/lti/__init__.py

```diff
@@ -34,6 +34,19 @@ def __init__(self, request, resource_id=None):
         self.request = request
         self._consumer_site = None
 
+    @property
+    def origin_url(self):
+        """Try to recreate the URL that was used to launch the LTI request."""
+        base_url = self.request.META.get("HTTP_REFERER")
```

Comment:
We have an already existing method `request_domain`. Is it possible to reuse it ?

Reply:
No, because `request_domain` is removing the protocol from the referrer.